tests/integration/bond_test.py

```diff
@@ -265,3 +265,81 @@ def test_set_bond_mac_address(eth1_up):
         eth1_cur_state = current_state[INTERFACES][1]
         assert bond99_cur_state[Interface.MAC] == MAC1.upper()
         assert bond99_cur_state[Interface.MAC] == eth1_cur_state[Interface.MAC]
+
+
+@pytest.fixture
+def bond99():
+    with bond_interface(name='bond99', slaves=[]) as bond_state:
+        yield bond_state
+
+
+def test_bond_and_test_inferfaces_have_different_macs(bond99):
+    slaves_state = statelib.show_only(('bond99', 'eth1', 'eth2'))
+    bond99_slaves_state = slaves_state[Interface.KEY][0]
+    eth1_slaves_state = slaves_state[Interface.KEY][1]
+    eth2_slaves_state = slaves_state[Interface.KEY][2]
+
+    bond99_slaves_mac = bond99_slaves_state[Interface.MAC]
+    eth1_slaves_mac = eth1_slaves_state[Interface.MAC]
+    eth2_slaves_mac = eth2_slaves_state[Interface.MAC]
+
+    assert bond99_slaves_mac != eth1_slaves_mac
+    assert eth1_slaves_mac != eth2_slaves_mac
+    assert eth2_slaves_mac != bond99_slaves_mac
+
+
+def test_bond_and_all_removed_slaves_have_same_mac(bond99):
+    bond99[INTERFACES][0]['link-aggregation']['slaves'] = []
+    libnmstate.apply(bond99)
+
+    slaves_state = statelib.show_only(('bond99', 'eth1', 'eth2'))
+    bond99_slaves_state = slaves_state[Interface.KEY][0]
+    eth1_slaves_state = slaves_state[Interface.KEY][1]
+    eth2_slaves_state = slaves_state[Interface.KEY][2]
+
+    bond99_slaves_mac = bond99_slaves_state[Interface.MAC]
+    eth1_slaves_mac = eth1_slaves_state[Interface.MAC]
+    eth2_slaves_mac = eth2_slaves_state[Interface.MAC]
+
+    assert bond99_slaves_mac != eth1_slaves_mac
+    assert eth1_slaves_mac != eth2_slaves_mac
+    assert eth2_slaves_mac != bond99_slaves_mac
+
+
+def test_bond_and_two_slaves_have_same_mac(bond99):
+    bond99[INTERFACES][0]['link-aggregation']['slaves'] = ['eth1', 'eth2']
+    libnmstate.apply(bond99)
+
+    slaves_state = statelib.show_only(('bond99', 'eth1', 'eth2'))
+    bond99_slaves_state = slaves_state[Interface.KEY][0]
+    eth1_slaves_state = slaves_state[Interface.KEY][1]
+    eth2_slaves_state = slaves_state[Interface.KEY][2]
+
+    prev_bond99_slaves_mac = bond99_slaves_state[Interface.MAC]
+    prev_eth1_slaves_mac = eth1_slaves_state[Interface.MAC]
+    prev_eth2_slaves_mac = eth2_slaves_state[Interface.MAC]
+
+    assert (prev_bond99_slaves_mac ==
+            prev_eth1_slaves_mac ==
+            prev_eth2_slaves_mac)
+
+    bond99[INTERFACES][0]['link-aggregation']['slaves'] = ['eth2', 'eth1']
```

Comment:
Better user `.reverse()`

Reply:
`>>> slaves = ['eth1', 'eth2']
>>> slaves
['eth1', 'eth2']
>>> slaves = ['eth1', 'eth2'].reverse()
>>> slaves
>>> print(slaves)
None
`

Since I cannot directly use the reverse() function, I used reversed(list_name)